src/providers/TransactionBuilderContextProvider.tsx

```diff
@@ -715,33 +704,108 @@ const TransactionBuilderContextProvider = ({
+    return dispatchedCrossChainActions?.filter((crossChainAction) =>
+      processingCrossChainActionIds.some((id) => id === crossChainAction.id)
     );
   }, [processingCrossChainActionIds, dispatchedCrossChainActions]);
 
-  const hasProcessingUnsent = useMemo(() => !!getFirstCrossChainActionByStatus(
-    dispatchedCrossChainActions ?? [],
-    CROSS_CHAIN_ACTION_STATUS.UNSENT,
-  ), [dispatchedCrossChainActions]);
+  const hasProcessingUnsent = useMemo(
+    () => !!getFirstCrossChainActionByStatus(dispatchedCrossChainActions ?? [], CROSS_CHAIN_ACTION_STATUS.UNSENT),
+    [dispatchedCrossChainActions]
+  );
 
   const [showMulticallOptions, setShowMulticallOptions] = useState<string | null>(null);
 
+  const onBuyClick = async () => {
+    const maticSdk = getSdkForChainId(CHAIN_ID.POLYGON);
+
+    if (!accountAddress || !maticSdk) return;
+
+    let account = await maticSdk.getAccount();
+    if (!account || account.address !== accountAddress) {
+      await maticSdk.computeContractAccount();
+      account = await maticSdk.getAccount();
+    }
+
+    if (!account) return;
+
+    let base64Hash = '';
+    let onRampOptions = {
+      lang: 'fr',
+      tab: 'buy',
+      tabs: 'buy',
+      chain: 'matic_mainnet',
+      net: 'matic_mainnet',
+      nets: 'arbitrum_mainnet,avalanche_mainnet,bsc_mainnet,fantom_mainnet,mainnet,optimism_mainnet,xdai_mainnet',
+      crys: 'AVAX,BNB,BTCB,BUSD,DAI,ETH,FRAX,LUSD,MAI,MATIC,RBTC,RDOC,RIF,USDC,USDT,WBTC,WETH,XCHF,XDAI,XTZ',
+      rfr: 'etherspot',
+      bsc: 'GBP',
+      bdc: 'MATIC',
+      mode: 'dark',
+      hash: '',
+      code: '',
+      addr: '',
+    };
+    const code = Math.floor(Math.random() * 8999) + 1000;
+    const message = 'MtPelerin-' + code;
+    onRampOptions.code = code.toString();
+    onRampOptions.addr = account.address || '';
+
+    if (account.state === AccountStates.UnDeployed) {
+      const submittedGateway = await deployAccount(maticSdk);
+      if (!submittedGateway) return;
+
+      maticSdk.notifications$
+        .pipe(
+          map(async (notification) => {
+            if (notification?.type === NotificationTypes.GatewayBatchUpdated) {
+              const submittedBatch = await maticSdk.getGatewaySubmittedBatch({
+                hash: submittedGateway.hash,
+              });
+
+              const failedStates = [
+                GatewayTransactionStates.Canceling,
+                GatewayTransactionStates.Canceled,
+                GatewayTransactionStates.Reverted,
+              ];
+
+              if (submittedBatch?.transaction?.state && failedStates.includes(submittedBatch?.transaction?.state)) {
+                console.log('error', 'Failed to deploy the etherspot wallet');
+              } else if (submittedBatch?.transaction?.state === GatewayTransactionStates.Sent) {
+                const signature = await maticSdk.signMessage({ message });
+                base64Hash = Buffer.from(signature.replace('0x', ''), 'hex').toString('base64');
```

Comment:
hmmm this won't continue on further code where it redirects to mt pelerin website because it's event based

Reply:
good catch, I was hoping that `base64Hash` would get overridden and used below, but forgot that this was a notification